src/shared/components/common/vote-buttons.tsx

```diff
@@ -0,0 +1,180 @@
+import { showScores } from "@utils/app";
+import { numToSI } from "@utils/helpers";
+import classNames from "classnames";
+import { Component, linkEvent } from "inferno";
+import { CommentAggregates, PostAggregates } from "lemmy-js-client";
+import { I18NextService } from "../../services";
+import { Icon, Spinner } from "../common/icon";
+import { PostListing } from "../post/post-listing";
+
+interface VoteButtonsProps {
+  postListing: PostListing;
+  enableDownvotes?: boolean;
+  upvoteLoading?: boolean;
+  downvoteLoading?: boolean;
+  handleUpvote: (i: PostListing) => void;
```

Comment:
What we've typically done, is pass the lemmy-js-client form (in this case `CreateCommentLike`) up to the parent. 

But here, since that could be either a post or a comment, I'd recommend passing up an id, the new vote, and an auth. Type the interface well, something like:

```
interface VoteItem {
  id: number,
  newVote: number,
  auth: ...
}
```

Reply:
Thanks -- are my latest edits more what you were picturing? I'm not sure where this `VoteItem` type could go.

Reply:
Comments use `comment_id` for their ID, and posts use `post_id`, so I had to add a `switch` and an extra property so comments/posts can specify which type to use when submitting the click.